### PR TITLE
Add e2e test to remove linux bridges

### PR DESCRIPTION
**Pull request by qinqon**
Using absent state linux bridge can be removed, so this
test that here, it also do a little refactor to reuse
checking of interfaces.

### CODE REVIEW

File: test/e2e/nodenetworkstate_test.go [hunk 1]
```diff
@@ -52,17 +52,33 @@ var _ = Describe("NodeNetworkState", func() {
...
-					}, ReadTimeout, ReadInterval).Should(ContainElement("br1"))
+			BeforeEach(func() {
+				createBridgeAtNodes("br1")
+				updateDesiredState(namespace, br1Absent)
+			})
+			AfterEach(func() {
+				// If not br1 is going to be removed if created manually
+				resetDesiredStateForNodes(namespace)
+			})
+			It("should have the linux bridge at currentState", func() {
```
Comment (phoracek) [inline test/e2e/nodenetworkstate_test.go]: wrong description? If linux bridge absent it should remove the linux bridge from host. Overall, I'd section it little diferently (just an idea for another PR).

```
Context("when there is a linux bridge configured on a node")
	// before each, create the bridge, after each, cleanup
	Context("and it is targeted as absent")
		// before each set desiredState
		It("should be removed from the host and reported state")
```

File: test/e2e/nodes_test.go [hunk 2]
```diff
@@ -47,15 +34,7 @@ var _ = Describe("Nodes", func() {
...
-					Eventually(func() []string {
-						var currentStateYaml nmstatev1.State
-						currentState(namespace, node, &currentStateYaml).ShouldNot(BeEmpty())
-
-						interfaces := interfaces(currentStateYaml)
-						Expect(interfaces).ToNot(BeEmpty(), "Node %s should have network interfaces", node)
-
-						return interfacesName(interfaces)
-					}, ReadTimeout, ReadInterval).Should(ContainElement(expectedDummyName))
+					interfacesForNode(node).Should(ContainElement("dummy0"))
```
[the remark this answers is not in the document]
Comment (qinqon) [reply test/e2e/nodes_test.go]: Done

File: test/e2e/nodes_test.go [hunk 3]
```diff
@@ -3,22 +3,13 @@ package e2e
...
 			for _, node := range nodes {
-				var currentStateYaml nmstatev1.State
-				currentState(namespace, node, &currentStateYaml).ShouldNot(BeEmpty())
-
-				interfaces := interfaces(currentStateYaml)
-				Expect(interfaces).ToNot(BeEmpty(), "Node %s should have network interfaces", node)
-
-				obtainedInterfaces := interfacesName(interfaces)
-				Expect(obtainedInterfaces).To(ContainElement("eth0"))
+				interfacesForNode(node).Should(ContainElement("eth0"))
```
[the remark this answers is not in the document]
Comment (qinqon) [reply test/e2e/nodes_test.go]: Ack, will move.

File: test/e2e/nodenetworkstate_test.go (hunk 1 above)
Comment (qinqon) [reply test/e2e/nodenetworkstate_test.go]: I think you are checking old stuff

Comment (phoracek) [reply test/e2e/nodenetworkstate_test.go]: Tests should be most of all easy to read. That I think means [Descriptions and organization of Contexts] > [Test Code Workflow] > [Test Code].

**Pull request merged**